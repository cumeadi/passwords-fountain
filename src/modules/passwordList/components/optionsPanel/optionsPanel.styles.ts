import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('section')`
    max-width: ${theme.breakpoints.s480};
    background: ${theme.colors.white};
    padding: ${theme.spacing.m18};
    box-shadow: ${theme.shadows.clickableItem};
`;

export const ContentWrapper = styled('div')``;

export const Content = styled('div')`
    padding-bottom: ${theme.spacing.m18};
`;

export const ButtonWrapper = styled('div')`
    display: grid;
    grid-auto-flow: column;
    grid-gap: ${theme.spacing.xl30};
`;
