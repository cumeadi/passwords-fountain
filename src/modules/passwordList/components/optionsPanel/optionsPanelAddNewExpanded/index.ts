export { OptionsPanelAddNewExpanded } from './optionsPanelAddNewExpanded.component';
