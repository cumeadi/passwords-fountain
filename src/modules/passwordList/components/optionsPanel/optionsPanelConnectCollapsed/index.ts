export { OptionsPanelConnectCollapsed } from './optionsPanelConnectCollapsed.component';
