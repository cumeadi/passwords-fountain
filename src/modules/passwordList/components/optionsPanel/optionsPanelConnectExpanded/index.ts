export { OptionsPanelConnectExpanded } from './optionsPanelConnectExpanded.component';
