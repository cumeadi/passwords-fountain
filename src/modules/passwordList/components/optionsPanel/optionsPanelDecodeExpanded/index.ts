export { OptionsPanelDecodeExpanded } from './optionsPanelDecodeExpanded.component';
