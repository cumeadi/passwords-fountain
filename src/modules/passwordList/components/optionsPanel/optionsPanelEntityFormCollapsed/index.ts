export { OptionsPanelEntityFormCollapsed } from './optionsPanelEntityFormCollapsed.component';
