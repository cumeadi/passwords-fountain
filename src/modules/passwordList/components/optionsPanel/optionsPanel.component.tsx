import { h } from 'preact';
import { useSelector, useAction } from '@/store';
import { TypedComponent } from '@/common/typings/prop-types';
import { Wrapper } from './optionsPanel.styles';
import { OptionsPanelConnectCollapsed } from './optionsPanelConnectCollapsed';
import { OptionsPanelConnectExpanded } from './optionsPanelConnectExpanded';
import { OptionsPanelEntityFormCollapsed } from './optionsPanelEntityFormCollapsed';
import { OptionsPanelEntityFormExpanded } from './optionsPanelEntityFormExpanded';
import { selectCurrentOptionPanelVariantName } from '@/modules/passwordList/passwordList.selectors';
import { passwordListActions } from '@/modules/passwordList/passwordList.actions';
import { VariantName } from '@/modules/passwordList/passwordList.contants';

type VariantSwitcher = (destination: VariantName) => void;

export interface VariantProps {
    switchCurrentVariantName: VariantSwitcher;
}

type VariantInstances = {
    [key in VariantName]: TypedComponent<VariantProps>;
};

export const OptionsPanel: TypedComponent<Props> = () => {
    const currentVariantName = useSelector(selectCurrentOptionPanelVariantName);
    const setCurrentVariantName = useAction(
        passwordListActions.switchOptionPanelVariant
    );

    const variants: VariantInstances = {
        connectCollapsed: OptionsPanelConnectCollapsed,
        connectExpanded: OptionsPanelConnectExpanded,
        entityFormCollapsed: OptionsPanelEntityFormCollapsed,
        entityFormExpanded: OptionsPanelEntityFormExpanded,
    };

    const CurrentVariant = variants[currentVariantName as VariantName];
    return (
        <Wrapper currentVariantName={currentVariantName}>
            <CurrentVariant switchCurrentVariantName={setCurrentVariantName} />
        </Wrapper>
    );
};

interface Props {}

OptionsPanel.propTypes = {};
