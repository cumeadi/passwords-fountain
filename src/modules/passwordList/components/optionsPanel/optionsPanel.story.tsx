import { h, VNode } from 'preact';
import { OptionsPanel } from './optionsPanel.component';
import { withA11y } from '@storybook/addon-a11y';
import { StoreProvider } from '@preact-hooks/unistore';
import createStore from 'unistore';

export default {
    title: 'OptionsPanel',
    decorators: [withA11y],
};
const storeMock = (variant: any): any =>
    createStore({
        passwordList: {
            currentOptionPanelVariantName: variant,
        },
    });

const withStoreMock = (render: () => VNode, variant: any): VNode => {
    const store = storeMock(variant);

    return <StoreProvider value={store}>{render()}</StoreProvider>;
};

export const defaultView = (): VNode =>
    withStoreMock(() => <OptionsPanel />, 'connectCollapsed');

export const connectExpanded = (): VNode =>
    withStoreMock(() => <OptionsPanel />, 'connectExpanded');

export const entityFormCollapsed = (): VNode =>
    withStoreMock(() => <OptionsPanel />, 'entityFormCollapsed');

export const entityFormExpanded = (): VNode =>
    withStoreMock(() => <OptionsPanel />, 'entityFormExpanded');
