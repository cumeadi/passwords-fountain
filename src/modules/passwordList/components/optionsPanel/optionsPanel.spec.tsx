import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { OptionsPanel } from './optionsPanel.component';
import { StoreProvider } from '@preact-hooks/unistore';
import createStore from 'unistore';

const stateMock = {
    passwordList: {
        currentOptionPanelVariantName: 'connectCollapsed',
    },
};

describe('OptionsPanel', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <StoreProvider value={createStore(stateMock)}>
                <OptionsPanel />
            </StoreProvider>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
