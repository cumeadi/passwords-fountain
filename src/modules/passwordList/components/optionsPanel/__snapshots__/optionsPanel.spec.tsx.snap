// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsPanel should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go1009640918"
    currentvariantname="connectCollapsed"
  >
    <section
      class="go115679"
    >
      <div
        class="go2997710935"
      >
        <button
          class="go3135434547"
          type="button"
        >
          <span
            class="go4062099715"
          >
            Settings
          </span>
        </button>
        <button
          class="go3135434547"
          type="button"
        >
          <span
            class="go4062099715"
          >
            Connect
          </span>
        </button>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
